examples/rust/tracecon/src/main.rs

```diff
@@ -0,0 +1,121 @@
+use anyhow::{anyhow, bail, Result};
+use core::time::Duration;
+use libbpf_rs::PerfBufferBuilder;
+use object::Object;
+use object::ObjectSymbol;
+use std::convert::TryInto;
+use std::fs;
+use std::net::Ipv4Addr;
+use std::path::Path;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+use structopt::StructOpt;
+
+#[path = "bpf/.output/tracecon.skel.rs"]
+mod tracecon;
+use tracecon::*;
+
+#[derive(Debug, StructOpt)]
+struct Command {
+    /// verbose output
+    #[structopt(long, short)]
+    verbose: bool,
+    /// glibc path
+    #[structopt(long, short, default_value = "/lib/x86_64-linux-gnu/libc.so.6")]
+    glibc: String,
+}
+
+fn bump_memlock_rlimit() -> Result<()> {
+    let rlimit = libc::rlimit {
+        rlim_cur: 128 << 20,
+        rlim_max: 128 << 20,
+    };
+
+    if unsafe { libc::setrlimit(libc::RLIMIT_MEMLOCK, &rlimit) } != 0 {
+        bail!("Failed to increase rlimit");
+    }
+
+    Ok(())
+}
+
+fn get_symbol_address(so_path: &str, fn_name: &str) -> Result<usize> {
+    let path = Path::new(so_path);
+    let buffer = fs::read(path)?;
+    let file = object::File::parse(buffer.as_slice())?;
+
+    let mut symbols = file.dynamic_symbols();
+    let symbol = symbols
+        .find(|symbol| {
+            if let Ok(name) = symbol.name() {
+                return name == fn_name;
+            }
+            false
+        })
+        .ok_or(anyhow!("symbol not found"))?;
+
+    Ok(symbol.address() as usize)
+}
+
+fn handle_event(_cpu: i32, data: &[u8]) {
+    // payload: 4 bytes tag + 4 bytes ip + 84 bytes hostname
```

Comment:
Hardcoding offsets isn't very nice. Maybe try using a global in the .bpf.c to trigger definition in skeleton. And then use generated type?

eg what runqslower does: https://github.com/libbpf/libbpf-rs/blob/4b5ee8dc147e781c4ba237efc7108c0f910b1963/examples/runqslower/src/bpf/runqslower.bpf.c#L13-L14

Reply:
Thanks, that's neat.